src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -125,20 +126,20 @@ private async Task<string> SetupInternalAsync(SetupContext context)
 
             // Due to database collation we normalize the userId to lower invariant.
             // During setup there are no users so we do not need to check unicity.
-            context.AdminUserId = _setupUserIdGenerator.GenerateUniqueId().ToLowerInvariant();
+            context.Properties[SetupConstants.AdminUserId] = _setupUserIdGenerator.GenerateUniqueId().ToLowerInvariant();
 
             var shellSettings = new ShellSettings(context.ShellSettings);
 
             if (string.IsNullOrEmpty(shellSettings["DatabaseProvider"]))
             {
-                shellSettings["DatabaseProvider"] = context.DatabaseProvider;
-                shellSettings["ConnectionString"] = context.DatabaseConnectionString;
-                shellSettings["TablePrefix"] = context.DatabaseTablePrefix;
+                shellSettings["DatabaseProvider"] = context.Properties[SetupConstants.DatabaseProvider]?.ToString();
+                shellSettings["ConnectionString"] = context.Properties[SetupConstants.DatabaseConnectionString]?.ToString();
+                shellSettings["TablePrefix"] = context.Properties[SetupConstants.DatabaseTablePrefix]?.ToString();
```

Comment:
as above TryGetValue. But here we should still set the shellSettings key even if there is no value provided I think @jtkech is it required that the shellsettings directionary have an empty key for `TablePrefix` ? Seems odd. But don't want to risk the breaking of anything

Reply:
It sets values in the memory config source of the shell settings that can hold null values

Then, as i remember in the code we check if the string is null or empty

So i think that's okay